arith/eulerian_polynomial_vec.c

```diff
@@ -0,0 +1,37 @@
+/*
+    Copyright (C) 2021 Albin Ahlbäck
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "arith.h"
+
+void
+arith_eulerian_polynomial_vec(fmpz_poly_t * res, const ulong n)
+{
+    ulong nx, mx;
+
+    fmpz_poly_one(res[0]);
+    if (n < 1)
+        return;
+
+    for (nx = 1; nx <= n; nx++)
+    {
+        fmpz_poly_zero(res[nx]);
+        fmpz_poly_set_coeff_si(res[nx], 0, 1);
+        fmpz_poly_set_coeff_si(res[nx], nx - 1, 1);
+        for (mx = 1; mx <= nx / 2; mx++)
+        {
+            fmpz_mul_ui(res[nx]->coeffs + mx,
+                    res[nx - 1]->coeffs + (mx - 1), nx - mx);
+            fmpz_addmul_ui(res[nx]->coeffs + mx,
+                    res[nx - 1]->coeffs + mx, mx + 1);
```

Comment:
when `mx = 1, nx = 2`, the access `res[nx - 1]->coeffs + mx` is an overrun because `res[1] is allocated to length 1` and `res[1]->coeffs + 1` is therefore an overrun.
```valgrind --leak-check=full ./build/arith/test/t-eulerian_polynomial_vec```

Reply:
BTW, I didn't mean to approve this, I just clicked the wrong button.